ocaml/forkexecd/lib/forkhelpers.ml

```diff
@@ -182,11 +182,13 @@ type syslog_stdout =
 
 (** Safe function which forks a command, closing all fds except a whitelist and
     having performed some fd operations in the child *)
-let safe_close_and_exec ?env stdin stdout stderr
+let safe_close_and_exec ?tracing ?env stdin stdout stderr
     (fds : (string * Unix.file_descr) list) ?(syslog_stdout = NoSyslogging)
     ?(redirect_stderr_to_stdout = false) (cmd : string) (args : string list) =
+  with_tracing ~tracing ~name:"safe_close_and_exec" @@ fun tracing ->
   let sock =
-    Fecomms.open_unix_domain_sock_client (runtime_path ^ "/xapi/forker/main")
+    Fecomms.open_unix_domain_sock_client ?tracing
+      (runtime_path ^ "/xapi/forker/main")
```

Comment:
It would be better to use `Filename.concat` to avoid issues like `....//...` in paths.

Reply:
https://github.com/xapi-project/xen-api/pull/5583/commits/e80041706a6aba5906379c921cd0370fe283bf13 Done!